src/CameraTarget.h

```diff
@@ -5,8 +5,23 @@
 class CCameraTarget : public CCameraPerspective {
 public:
 	// --- PUT YOUR CODE HERE ---
-	//CCameraTarget(Size resolution, const Vec3f& pos, const Vec3f& target, const Vec3f& up, float angle) ...
-	
+	CCameraTarget(Size resolution, const Vec3f& pos, const Vec3f& target, const Vec3f& up, float angle) :
+	    CCameraPerspective(resolution, pos, normalize(target - pos), up, angle) {
+	        m_target = target;
+	}
+
+    virtual void setPosition(const Vec3f& pos) override {
+	    m_pos = pos;
+	}
+
+	virtual void setTarget(const Vec3f target) {
+	    m_target = target;
+	}
+
+	Vec3f getTarget(void) {
+	    return m_target;
+	}
 private:
 	Vec3f m_target;		///< Camera target point in WCS
-}
+	Vec3f m_pos;
```

Comment:
position must not be here
setPosition() and setTarget() both affect the direction of the camera